spec/models/chargeback_configured_system_spec.rb

```diff
@@ -0,0 +1,111 @@
+RSpec.describe ChargebackConfiguredSystem do
+  include Spec::Support::ChargebackHelper
+
+  let(:admin) { FactoryBot.create(:user_admin) }
+  let(:base_options) do
+    {:interval_size       => 2,
+     :end_interval_offset => 0,
+     :tag                 => '/managed/environment/prod',
+     :ext_options         => {:tz => 'UTC'},
+     :userid              => admin.userid}
+  end
+  let(:hourly_rate)               { 0.01 }
+  let(:count_hourly_rate)         { 1.00 }
+  let(:cpu_count)                 { 1.0 }
+  let(:memory_available)          { 1000.0 }
+  let(:starting_date) { Time.parse('2012-09-01 23:59:59Z').utc }
+  let(:ts) { starting_date.in_time_zone(Metric::Helper.get_time_zone(base_options[:ext_options])) }
+  let(:report_run_time) { month_end }
+  let(:month_beginning) { ts.beginning_of_month.utc }
+  let(:month_end) { ts.end_of_month.utc }
+  let(:hours_in_month) { Time.days_in_month(month_beginning.month, month_beginning.year) * 24 }
+  let(:ems) { FactoryBot.create(:ems_vmware) }
+
+  let(:hourly_variable_tier_rate)       { {:variable_rate => hourly_rate.to_s} }
+  let(:count_hourly_variable_tier_rate) { {:variable_rate => count_hourly_rate.to_s} }
+  let(:fixed_compute_tier_rate)         { {:fixed_rate => hourly_rate.to_s} }
+
+  let(:detail_params) do
+    {
+      :chargeback_rate_detail_cpu_allocated      => {:tiers => [count_hourly_variable_tier_rate]},
+      :chargeback_rate_detail_memory_allocated   => {:tiers => [hourly_variable_tier_rate]},
+      :chargeback_rate_detail_fixed_compute_cost => {:tiers => [fixed_compute_tier_rate]}
+    }
+  end
+
+  let!(:chargeback_rate) do
+    FactoryBot.create(:chargeback_rate, :detail_params => detail_params)
+  end
+
+  before do
+    MiqRegion.seed
+    ChargebackRateDetailMeasure.seed
+    ChargeableField.seed
+    ManageIQ::Showback::InputMeasure.seed
+    MiqEnterprise.seed
+
+    EvmSpecHelper.create_guid_miq_server_zone
+    cat = FactoryBot.create(:classification, :description => "Environment", :name => "environment", :single_value => true, :show => true)
+    c = FactoryBot.create(:classification, :name => "prod", :description => "Production", :parent_id => cat.id)
+    @tag = Tag.find_by(:name => "/managed/environment/prod")
+    ChargebackRate.set_assignments(:compute, [{:cb_rate => chargeback_rate, :tag => [c, "vm"]}])
```

Comment:
@k-chu 

this type of assignment is not relevant for chargeback for configured system and it works by accident because we have to change method `ConsumptionWithoutRollup#tag_list_with_prefix` to reflect also configured systems, possible solution:


```diff
diff --git a/app/models/chargeback/consumption_without_rollups.rb b/app/models/chargeback/consumption_without_rollups.rb
index 0552fa020b..bc7e334c07 100644
--- a/app/models/chargeback/consumption_without_rollups.rb
+++ b/app/models/chargeback/consumption_without_rollups.rb
@@ -24,8 +24,17 @@ class Chargeback
       resource.id
     end


+    def tag_prefix
+      klass_prefix = case resource
+                     when VmOrTemplate     then 'vm'
+                     when ConfiguredSystem then 'configured_system'
+                     end
+
+      "#{klass_prefix}/tag/managed/"
+    end
+
     def tag_list_with_prefix
-      tag_names.map { |t| "vm/tag/managed/#{t}" }
+      tag_names.map { |t| "#{tag_prefix}#{t}" }
     end

     def parents_determining_rate

```

and you need  to assign rate to MiqEnterprise in spec:

```ruby
ChargebackRate.set_assignments(:compute, [{:cb_rate => chargeback_rate, :object => MiqEnterprise.first}])
```

Reply:
Addressed this with your suggestions. Thanks for the feedback.